API/testrunner/devices/artik530.py

```diff
@@ -0,0 +1,137 @@
+        '''
+        if not self.workdir:
+            console.fail('Please use --remote-workdir for the device.')
+        if not self.address:
+            console.fail('Please define the IP address of the device.')
+        if not self.user:
+            console.fail('Please define the username of the device.')
+
+        if self.workdir is '/':
+            console.fail('Please do not use the root folder as test folder.')
+
+    def initialize(self):
+        '''
+        Flash the device.
+        '''
+        if self.env['info']['no_flash']:
+            return
+        
+        target_app = self.env['modules']['app']
+        build_path = self.env['paths']['build']
+
+        test_src = target_app['paths']['tests']
+        test_dst = utils.join(build_path, 'test')
+
+        # 1. Copy all the necessary files.
+        # Copy applicaiton RPM package file.
+        rpm_package_path = self.env['paths']['tizen-rpm-package']
+        utils.copy(rpm_package_path, build_path)
+
+        # Copy all the tests into the build folder.
+        utils.copy(test_src, test_dst)
+        utils.copy(paths.SIMPLE_TESTER, build_path)
+
+        # 2. Deploy the build folder to the device.
+        rsync_flags = ['--recursive', '--compress', '--delete']
+
+        # Note: slash character is required after the path.
+        # In this case `rsync` copies the whole folder, not
+        # the subcontents to the destination.
+        src = self.env['paths']['build'] + '/'
+        dst = '%s@%s:%s' % (self.user, self.address, self.workdir)
+
+        utils.execute('.', 'rsync', rsync_flags + [src, dst])
+
+        # 3. Install rpm package
+        self.login()
+        self.channel.exec_command('mount -o remount,rw /')
+        template = 'rpm -ivh --force --nodeps %s/%s-1.0.0-0.armv7l.rpm'
+        self.channel.exec_command(template % (self.workdir, self.app))
+        
+        self.logout()
+
+    def login(self):
+        '''
+        Login to the device.
+        '''
+        self.channel.open()
+
+    def logout(self):
+        '''
+        Logout from the device.
+        '''
+        self.channel.close()
+
+    def execute(self, testset, test):
+        '''
+        Execute the given test.
+        '''
+        self.login()
+
+        template = 'python3 %s/simpletester.py --cwd %s --cmd %s --testfile %s'
+        # Absolute path to the test folder.
+        testdir = '%s/test' % self.workdir
+        # Absolute path to the test file.
+        testfile = '%s/%s/%s' % (testdir, testset, test['name'])
+
+        # Create the command that the device will execute.
+        command = template % (self.workdir, testdir, self.app, testfile)
+
+        stdout = self.channel.exec_command(command)
+
+        # Since the stdout is a JSON text, parse it.
+        result = json.loads(stdout)
+        # Make HTML friendly stdout.
+        result['output'] = result['output'].rstrip('\n').replace('\n', '<br>')
+
+        result['memstat'] = -1 # TODO
```

Comment:
You can remove this line if you already have memstat member (with 'n/a' values) in the result object.